### PR TITLE
fix(expressions): quote bare words inside a oneOf array literal

### PR DESCRIPTION
**Pull request by hawkeyexl**
## What changed

Fixes the root cause of the `oneOf` quoting gotcha noted in [#585](https://github.com/doc-detective/doc-detective/pull/585): `$$platform oneOf [linux, mac, windows]` (bare, unquoted array elements) silently evaluated to `false`/undefined instead of matching.

## Root cause

`preprocessExpression` (`src/core/expressions.ts`) quotes `oneOf`'s **left** operand via `quoteIfLiteral`, but spliced the right-hand array literal into the generated function body verbatim. `[linux, mac, windows]` compiled to three unquoted JS identifiers with no binding in scope, throwing a `ReferenceError` inside the generated function. That error is caught by `evaluateExpression`'s try/catch and swallowed to `undefined`, so the condition failed closed with no diagnostic — `contains` doesn't have this problem because its single RHS operand already goes through `quoteIfLiteral`.

## Fix

Added a `quoteArrayItems` helper that recognizes an actual `[...]` array literal, splits it on top-level commas (bracket-depth aware, so a nested literal item isn't split mid-way), and runs each element through the existing `quoteIfLiteral` before rejoining — so bare words, numbers, booleans, and already-quoted strings inside the array are handled the same way `contains`'s operand already is. A bare `$$var` RHS (not a literal array) is left untouched, since `replaceMetaValues` already resolves it to valid JSON before this code runs.

## Testing

- 4 new red→green unit tests in `test/expressions-unit.test.js` (bare-word true/false, mixed quoted/bare, numeric-array regression guard) — confirmed failing before the fix, passing after.
- New fixture step in `test/core-artifacts/interactions/custom-assertions.spec.json` exercising `$$platform oneOf [linux, mac, windows]` end-to-end through the real runner; verified PASS via `doc-detective runTests`.
- Full `test/expressions-unit.test.js` + `test/expressions-coverage.test.js` (123 tests) pass.
- Broader assertion/routing suites (`checkLink-assertions`, `custom-assertions`, `httpRequest-assertions`, `routing-context`, `runCode-assertions`, `runShell`, `core-core`) pass unchanged (147 tests, 0 failures).

## Docs impact

This is a bug fix to documented expression syntax, so it carries a docs update: added a bare-word `oneOf` example to `docs/fern/pages/reference/expressions.mdx` and cross-linked the existing string-literal quoting rule, per the follow-up note left in #585.

## ADR

Per repo convention, this behavior change ships with [`adrs/01052-quote-bare-words-in-oneof-array-literals.md`](adrs/01052-quote-bare-words-in-oneof-array-literals.md).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed `oneOf` handling for literal array options written with bare words (e.g., `$$platform oneOf [linux, mac, windows]`) so options are treated as strings and evaluate reliably.
  * Ensured mixed quoted/bare entries work as expected, while preserving existing behavior for numeric/boolean/null and other non-literal RHS forms.
* **Documentation**
  * Updated `oneOf` docs to clarify supported value types and string quoting rules (including values that resemble numbers/booleans/null).
* **Tests**
  * Expanded unit and end-to-end coverage for bare-word, mixed, and numeric arrays, including trailing-comma and “hole” parsing edge cases.

### CODE REVIEW

File: src/core/expressions.ts [hunk 1]
```diff
@@ -543,6 +543,41 @@ function preprocessExpression(expression: string): string {
...
+      if (char === "," && depth === 0) {
+        items.push(current);
+        current = "";
+      } else {
+        current += char;
+      }
+    }
+    items.push(current);
+    return `[${items.map((item) => quoteIfLiteral(item.trim())).join(", ")}]`;
+  };
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/core/expressions.ts:588]: Fixed: `quoteArrayItems` now drops any segment that's empty after trimming (a trailing comma like `[0, 1,]` or a hole like `[a,,b]`) instead of quoting it into a spurious `""` element, matching real JS array-literal elision. Added regression tests for both cases in `test/expressions-unit.test.js`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/fern/pages/reference/expressions.mdx [hunk 2]
```diff
@@ -79,10 +79,11 @@ Write word operators between the operands, with a space on each side.
   $$response.body.role contains "admin"
   ```
 
-- **`oneOf`**: True when the left operand equals one of the values in the array on the right.
+- **`oneOf`**: True when the left operand equals one of the values in the array on the right. The array can hold numbers or strings; the same [string literal](#string-literals) quoting rule applies to each value, so bare words work unless they contain spaces or operator characters.
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply docs/fern/pages/reference/expressions.mdx]: Reworded: the array splits on top-level commas, not whitespace, so a bare word with spaces already worked (the general string-literal 'quote if it has spaces or operator characters' rule doesn't apply here). The doc now states the actual constraint—quote a string value only if it needs to contain a comma or would otherwise look like a number/`true`/`false`/`null`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: adrs/01052-quote-bare-words-in-oneof-array-literals.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+# `oneOf` must quote bare words inside its array-literal operand
+
+## Context and Problem Statement
+
+The `oneOf` word operator (`src/core/expressions.ts`, `preprocessExpression`) rewrites
+`<value> oneOf <options>` into a call to the runtime `oneOf(value, options)` helper. The left
+operand is passed through `quoteIfLiteral`, which wraps a bare identifier in quotes unless it's
+already a string/number/boolean/array/object literal. The right operand — the options array — was
+never processed at all; it was spliced into the generated function body verbatim
+(`` `oneOf(${quoteIfLiteral(left)}, ${right.trim()})` ``).
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply adrs/01052-quote-bare-words-in-oneof-array-literals.md]: Fixed: simplified to a single inline code span (removed the redundant nested single-backtick pair inside the double-backtick escape).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/fern/pages/reference/expressions.mdx [hunk 4]
```diff
@@ -79,10 +79,11 @@ Write word operators between the operands, with a space on each side.
   $$response.body.role contains "admin"
   ```
 
-- **`oneOf`**: True when the left operand equals one of the values in the array on the right.
+- **`oneOf`**: True when the left operand equals one of the values in the array on the right. The array can hold numbers or strings; write string values bare unless one needs to contain a comma (the array's separator) or needs to look like a number, `true`, `false`, or `null`—quote those with the same [string literal](#string-literals) rules used elsewhere.
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply docs/fern/pages/reference/expressions.mdx]: Fixed: reworded to `The array can hold numbers, `true`, `false`, `null`, or strings` so booleans/null are explicitly listed as supported literal values, not just quoting hazards. Left `undefined`/`NaN`/`Infinity` out of the user-facing list—they're accepted internally by `quoteIfLiteral` but aren't meaningful values to compare against in a doc test, so documenting them would add noise without helping a reader.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
